src/app/(auth)/reset-password/page.tsx

```diff
@@ -28,15 +29,22 @@ export default function ResetPassword() {
     });
   }, []);
 
-  const resetPassword = async () => {
+  const handleResetPassword = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
     if (!passwordComplexity) {
-      setErrorMessage('Password must meet complexity requirements');
+      setErrorMessage('Password must meet complexity requirements.');
       return;
     }
     if (newPassword !== newPassword2) {
       setErrorMessage('Passwords do not match.');
       return;
     }
+    if (!verifyUserPassword(newPassword)) {
+      setErrorMessage(
+        'Password cannot be the same as your previous password. Please choose a different password.',
+      );
+      return;
+    }
     setErrorMessage('');
     const { error } = await supabase.auth.updateUser({ password: newPassword });
     if (error) {
```

Comment:
similar case here: we may want to just say "Something went wrong. Please try again later."